### PR TITLE
Add endpoint to copy tags from source repository

### PR DESCRIPTION
**Pull request by asmacdo**
https://pulp.plan.io/issues/4947
fixes #4947

### CODE REVIEW

**Comment by asmacdo** [comment]
While working on this, I noticed that @lubosmj and I both forgot to include request parameters for the REST API docs. Since it was a very minor change, I just included it here.

File: pulp_docker/app/serializers.py [hunk 1]
```diff
@@ -290,3 +290,49 @@ class DockerRecursiveAddSerializer(serializers.Serializer):
...
         required=False
     )
+
+
+class TagCopySerializer(serializers.Serializer):
+    """
+    Serializer for copying tags from a source repository to a destination repository.
+    """
+
+    source_repository = serializers.HyperlinkedRelatedField(
```
Comment (ipanova) [inline pulp_docker/app/serializers.py]: @asmacdo do you think you can add also source_repo_version?

File: pulp_docker/app/viewsets.py [hunk 2]
```diff
@@ -280,3 +285,46 @@ def create(self, request):
...
+        tags_in_repo = models.Tag.objects.filter(
+            pk__in=content_tags_in_repo,
+        )
+        if names is None:
+            tags_to_add = tags_in_repo
+        else:
+            tags_to_add = tags_in_repo.filter(name__in=names)
+
+        result = enqueue_with_reservation(
+            tasks.recursive_add_content, [destination],
```
Comment (ipanova) [inline pulp_docker/app/viewsets.py]: i think we should also lock source_repo, especially if it looks at latest version, which can change

File: pulp_docker/app/serializers.py (hunk 1 above)
Comment (asmacdo) [reply pulp_docker/app/serializers.py]: Sure

File: pulp_docker/app/viewsets.py (hunk 2 above)
Comment (asmacdo) [reply pulp_docker/app/viewsets.py]: On second thought, there is no need to reserve on the source repo. The viewset filters the content of the source repository synchronously, and the recursive_add task is given a list of content hrefs, which are independent of the source repo.

File: pulp_docker/app/serializers.py [hunk 3]
```diff
@@ -290,3 +292,68 @@ class DockerRecursiveAddSerializer(serializers.Serializer):
...
+        queryset=Repository.objects.all(),
+        view_name='repositories-detail',
+        label=_('Repository'),
+        error_messages={
+            'required': _('Destination repository URI must be specified.')
+        }
+    )
+    names = serializers.ListField(
+        required=False,
+        allow_null=True,
```
Comment (ipanova) [inline pulp_docker/app/serializers.py]: s/True/False

**Pull request merged**